data/scripts/variants_spdi.py

```diff
@@ -78,13 +73,38 @@ def convert_spdi(spdi, seq):
     return spdi
 
 
+def build_hgvs_from_spdi(spdi):
+    ins_seq = spdi.split(':')[-1]
+    del_seq = spdi.split(':')[2]
+    spdi_pos = int(spdi.split(':')[1])
+    chr_ref = spdi.split(':')[0]
+    if len(ins_seq) == 1 and len(del_seq) == 1:
+        hgvs = f'{chr_ref}:g.{spdi_pos + 1}{del_seq}>{ins_seq}'
+    # check if this variant is a deletion
+    elif len(del_seq) > len(ins_seq):
+        pos_hgvs_start = spdi_pos + 1 + len(ins_seq)
+        pos_hgvs_end = spdi_pos + len(del_seq)
+        if pos_hgvs_start == pos_hgvs_end:
+            hgvs = f'{chr_ref}:g.{pos_hgvs_start}del'
+        else:
+            hgvs = f'{chr_ref}:g.{pos_hgvs_start}_{pos_hgvs_end}del'
+    # it is a insertion. We will not generate specail case hgvs - duplication, since using insertion for dup is valid.
+    # you can check if the hgvs is valid here: https://api.ncbi.nlm.nih.gov/variation/v0/#/HGVS/get_hgvs__hgvs__contextuals
+    else:
+        pos_hgvs_start = spdi_pos + len(del_seq)
+        pos_hgvs_end = spdi_pos + len(del_seq) + 1
+        insert_seq_hgvs = ins_seq[len(del_seq):]
+        hgvs = f'{chr_ref}:g.{pos_hgvs_start}_{pos_hgvs_end}ins{insert_seq_hgvs}'
+    return hgvs
```

Comment:
We will also need to catch `delins` type, I think the main logic will be to check if `del_seq` is a subset of `ins_seq` or in the other way around.